code/addons/docs/src/manifest.ts

```diff
@@ -0,0 +1,185 @@
+import * as fs from 'node:fs/promises';
+import * as path from 'node:path';
+
+import { groupBy } from 'storybook/internal/common';
+import { logger } from 'storybook/internal/node-logger';
+import type {
+  ComponentManifest,
+  DocsIndexEntry,
+  IndexEntry,
+  Manifests,
+  Path,
+  PresetPropertyFn,
+  StorybookConfigRaw,
+} from 'storybook/internal/types';
+
+const ATTACHED_MDX_TAG = 'attached-mdx';
+const UNATTACHED_MDX_TAG = 'unattached-mdx';
+
+export interface DocsManifestEntry {
+  id: string;
+  name: string;
+  path: Path;
+  title: string;
+  content?: string;
+  error?: { name: string; message: string };
+}
+
+interface DocsManifest {
+  v: number;
+  docs: Record<string, DocsManifestEntry>;
+}
+
+interface ComponentManifestWithDocs extends ComponentManifest {
+  docs?: Record<string, DocsManifestEntry>;
+}
+
+interface ComponentsManifestWithDocs {
+  v: number;
+  components: Record<string, ComponentManifestWithDocs>;
+}
+
+interface ManifestsWithDocs extends Manifests {
+  docs?: DocsManifest;
+  components?: ComponentsManifestWithDocs;
+}
+
+/** Converts a DocsIndexEntry to a DocsManifestEntry by reading its file content. */
+export async function createDocsManifestEntry(entry: DocsIndexEntry): Promise<DocsManifestEntry> {
+  const absolutePath = path.join(process.cwd(), entry.importPath);
```

Comment:
As discussed, evaluating the absolute path of the import path needs a helper function and refactoring, because it's opaque to know what the "workingDir" actually is. I have agreed with Jeppe that refactoring and adjusting other code places as well will be a separate task, likely implemented in the next cooldown cycle.